### DISCUSSION TITLE
How to start training from an existing model in a different language than the target language

### DISCUSSION THREAD

**Discussion by Ca-ressemble-a-du-fake** [coqui-ai/TTS#1860] [General Q&A] [answered]
Hi,

I am trying to use @thorstenMueller 's VITS model in German as a starting point for my French model (I believe this is called "transfer learning").

So I cloned the repo and then followed the tutorial "[Explaining Checkpoints and restore/continue possibilities with COquiTTS](https://youtu.be/O6KxJR95WpE?list=PL19C7uchWZerUT0qIiEv7m2zXBs5kYl1L&t=878)" and first settled in the folder containing [@thorstenMueller 's model](https://github.com/coqui-ai/TTS/releases/download/v0.7.0_models/tts_models--de--thorsten--vits.zip), then defined config.json with a French dataset and language, alphabets ... and finally called the training_tts.py :

`CUDA_VISIBLE_DEVICES="0" python3 /home/caraduf/YourTTS/TTS/TTS/bin/train_tts.py --config_path /home/caraduf/dataset1/config_VITS.json --restore_path  model_file.pth`

But I got :

`
| > Found 563 files in /home/caraduf/dataset1
 > Using model: vits
 > Setting up Audio Processor...
 | > sample_rate:22050
 | > resample:False
 | > num_mels:80
 | > log_func:np.log10
 | > min_level_db:-100
 | > frame_shift_ms:None
 | > frame_length_ms:None
 | > ref_level_db:20
 | > fft_size:1024
 | > power:1.5
 | > preemphasis:0.0
 | > griffin_lim_iters:60
 | > signal_norm:True
 | > symmetric_norm:True
 | > mel_fmin:0
 | > mel_fmax:None
 | > pitch_fmin:0.0
 | > pitch_fmax:640.0
 | > spec_gain:20.0
 | > stft_pad_mode:reflect
 | > max_norm:4.0
 | > clip_norm:True
 | > do_trim_silence:True
 | > trim_db:45
 | > do_sound_norm:False
 | > do_amp_to_db_linear:True
 | > do_amp_to_db_mel:True
 | > do_rms_norm:False
 | > db_level:None
 | > stats_path:None
 | > base:10
 | > hop_length:256
 | > win_length:1024
Traceback (most recent call last):
  File "/home/caraduf/YourTTS/TTS/TTS/bin/train_tts.py", line 71, in <module>
    main()
  File "/home/caraduf/YourTTS/TTS/TTS/bin/train_tts.py", line 58, in main
    trainer = Trainer(
  File "/home/caraduf/.local/lib/python3.8/site-packages/trainer/trainer.py", line 372, in __init__
    output_path = get_experiment_folder_path(config.output_path, config.run_name)
  File "/home/caraduf/.local/lib/python3.8/site-packages/trainer/generic_utils.py", line 54, in get_experiment_folder_path
    output_folder = os.path.join(root_path, model_name + "-" + date_str + "-" + commit_hash)
  File "/usr/lib/python3.8/posixpath.py", line 76, in join
    a = os.fspath(a)
`

I also tried with the English VITS model that comes with TTS (see ~/.local/share/tts/tts_models--en--ljspeech--vits) with the same error message.

Should I add a line in my custom config.json ?

Edit : the variable `root_path` is None. So this is why the join function reports the error. 

Any help appreciated :) !

**Comment by Ca-ressemble-a-du-fake** [comment, marked as answer]
Actually `config.output_path` was `None` because I missed to define "output_path" in "config.json" file as required per https://github.com/coqui-ai/TTS/blob/f237e4ccd9f2fd2b0cb5e136dfe0e20cc32bf898/docs/source/tutorial_for_nervous_beginners.md#cli-way

**Reply by thorstenMueller** [reply]
Happy that you found/shared the solution 👍 .
